plugins/zstack.py

```diff
@@ -372,64 +389,87 @@ def acquire(self, dlg):
         stream_paused = str_ctrl.pauseStreams()
         dlg.pauseSettings()
 
-        nb = self.numberofAcquisitions.value
+        nb = self.numberOfAcquisitions.value
         ss = self._get_acq_streams()
-
         sacqt = acqmng.estimateTime(ss)
+        logging.debug("Acquisition streams: %s", ss)
+
+        # all_ss is a list of list of streams to acquire. In theory, we could do
+        # several set of acquisitions with each a set of streams. However, that's
+        # not how it's used. It's just a generic way to handle both cases:
+        # either each acquisition has only one stream, or there is a single
+        # acquisition to do all the stream.
+        if self._streams_intertwined:
+            # Streams are fastest changed: for each step, all streams are acquired
+            all_ss = [ss]
+        else:
+            # Streams are slowest changed: for each stream, do all steps together
+            all_ss = [[s] for s in ss]
         
-        completed = False
-
         try:
-            step_time = self.initAcquisition()
-            logging.debug("Acquisition streams: %s", ss)
+            # list of list of list of DataArray: for each acquisition, for each stream, for each step, the data acquired
+            all_images = [[] for _ in all_ss]
```

Comment:
Is this the same as `[[],] * len(all_ss) `?

Reply:
It's not. It's actually an easy mistake in Python. `l = [[]] * 3` returns a list containing 3 times the same reference to the same (sub)list. This means that if you do `l[1].append(0)`, you'll get `[[0], [0], [0]]`. 
That's not the case with the for loop.